### PR TITLE
feat: add tasks module

### PR DESCRIPTION
**Pull request by Leon-pix-coder**
Add tasks module

### CODE REVIEW

File: src/components/InviteFriends/index.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      `${didWallet?.walletInfo.address}-${timestamp}`
+    ).toString("hex");
+    const signature = AElf.wallet
+      .sign(message, didWallet?.walletInfo.keyPair)
+      .toString("hex");
+    const publicKey = wallet?.extraInfo?.publicKey ?? "";
+    const { caHash } = await getCaHashAndOriginChainIdByWallet(
+      wallet,
+      walletType
+    );
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx]: const { portkeyInfo: { walletInfo }, publicKey } = wallet?.extraInfo || {};

deconstruct your object like this to make the code more readable

File: src/components/InviteFriends/index.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      body: new URLSearchParams(requestObject).toString(),
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+    const portKeyResObj = await portKeyRes.json();
+    if (portKeyRes.ok) {
+      const token = portKeyResObj.access_token;
+      const res = await fetch(
+        portkeyServer + "/api/app/growth/shortLink?projectCode=13027",
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx]: make project code a constants

File: src/components/InviteFriends/index.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        {
+          method: "GET",
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx]: default is GET, you should not need to put GET here

File: src/components/InviteFriends/index.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    const portKeyRes = await fetch(connectUrl + "/connect/token", {
+      method: "POST",
+      body: new URLSearchParams(requestObject).toString(),
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+    const portKeyResObj = await portKeyRes.json();
+    if (portKeyRes.ok) {
+      const token = portKeyResObj.access_token;
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx:72]: what if access_token is undefined? do you still proceed?

File: src/components/InviteFriends/index.tsx [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    }
+    return null;
+  };
+
+  const handleCopy = () => {
+    setCopied(tgLinkWithCode);
+    setIsCopied(true);
+  };
+
+  const { data: referralCodeRes, loading } = useRequest(async () => {
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx]: if you are not using await, you can useRequest(generateCode)

File: src/components/InviteFriends/index.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+                </div>
+              </div>
+
+              <button
+                className="w-full h-[40px] bg-primary text-white font-bold text-[14px] font-outfit rounded-[24px]"
+                type="button"
+                onClick={() => {
+                  if (window?.Telegram?.WebApp?.openTelegramLink) {
+                    window?.Telegram?.WebApp?.openTelegramLink(
+                      `https://t.me/share/url?url=${tgLinkWithCode}`
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx:204]: what is this link?

File: src/components/InviteFriends/index.tsx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+          {loading ? (
+            <Loading className="h=[120px]" />
+          ) : (
+            <>
+              <div className="w-[160px] h-[160px] overflow-hidden rounded-[12px]">
+                <QRCode
+                  value={tgLinkWithCode}
+                  size={136}
+                  quietZone={12}
+                  logoImage="https://cdn.tmrwdao.com/votigram/assets/imgs/8489A36AA76C.webp"
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx:166]: why is there a portkey logo here?

File: src/components/InviteFriends/index.tsx [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+            className="w-full  bg-primary rounded-[24px] text-[14px] font-bold py-[10px] font-outfit leading-[14px] text-white"
+          >
+            Invite friends
+          </button>
+        </div>
+      </div>
+      <Drawer
+        isVisible={isShowDrawer}
+        direction="bottom"
+        canClose={true}
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx]: if its true, you do not need to pass true

File: src/components/InviteFriends/index.tsx [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+            <div className="flex-1 h-[10px] shrink-0 bg-white rounded-[5px]">
+              <div
+                className={clsx(
+                  "h-[10px] bg-gradient-to-r from-lime-green to-lime-primary rounded-[5px]"
+                )}
+                style={{ width: `${progress}%` }}
+              />
+            </div>
+            <span className="font-outfit font-bold text-[14px] leading-[14px] text-white shrink-0">
+              {data?.votigramVoteAll || 0}/{data?.totalInvitesNeeded || 20}
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx:134]: what if totalInvitesNeeded is undefined?

File: src/components/InviteFriends/index.tsx (hunk 6 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx:204]: This part refers to tomorrowDao

File: src/components/InviteFriends/index.tsx (hunk 7 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx:166]: Figma design like this

File: src/components/InviteFriends/index.tsx (hunk 9 above)
Comment (evan-spidy) [reply src/components/InviteFriends/index.tsx:134]: ignore this. I misread the code

File: src/components/InviteFriends/index.tsx (hunk 8 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx]: Updated

File: src/components/InviteFriends/index.tsx (hunk 5 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx]: Updated

File: src/components/InviteFriends/index.tsx (hunk 4 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx:72]: Updated

File: src/components/InviteFriends/index.tsx (hunk 3 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx]: Updated

File: src/components/InviteFriends/index.tsx (hunk 2 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx]: Updated

File: src/components/InviteFriends/index.tsx (hunk 1 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx]: Updated

**Pull request merged**